src/PrestaShopBundle/Controller/Admin/Improve/Design/ThemeController.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Controller\Admin\Improve\Design;
+
+use PrestaShop\PrestaShop\Core\Domain\Meta\DataTransferObject\LayoutCustomizationPage;
+use PrestaShop\PrestaShop\Core\Domain\Meta\Query\GetPagesForLayoutCustomization;
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController as AbstractAdminController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Class ThemeController manages "Improve > Design > Theme & Logo" pages.
+ */
+class ThemeController extends AbstractAdminController
+{
+    /**
+     * Show Front Office theme's pages layout customization.
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function customizePageLayoutsAction(Request $request)
+    {
+        /** @var LayoutCustomizationPage[] $pages */
+        $pages = $this->getQueryBus()->handle(new GetPagesForLayoutCustomization());
+
+        $pageLayoutCustomizationFormFactory =
+            $this->get('prestashop.bundle.form.admin.improve.design.theme.page_layout_customization_form_factory');
+        $pageLayoutCustomizationForm = $pageLayoutCustomizationFormFactory->create($pages);
+        $pageLayoutCustomizationForm->handleRequest($request);
+
+        if ($pageLayoutCustomizationForm->isSubmitted()) {
+            $themePageLayoutsCustomizer = $this->get('prestashop.core.addon.theme.theme.page_layouts_customizer');
+            $themePageLayoutsCustomizer->customize($pageLayoutCustomizationForm->getData()['layouts']);
+
+            \Tools::clearCache();
```

Comment:
hopefully i can replace legacy call by service when this PR gets merged => https://github.com/PrestaShop/PrestaShop/pull/10770 since i dont want to add another method without interface to `CacheClearer`. :/

Reply:
👍

Reply:
i think we cannot merge this PR, until #10770 is merged?

Reply:
Yes, we wait for @jolelievre who's at PS Day Warsaw? I'll urge him Monday ;)